src/src/Controller/AbandonedCartController.php

```diff
@@ -0,0 +1,254 @@
+        }
+    }
+
+    public function sync(string $token): bool
+    {
+        if ($this->configuration->get('SMAILY_CART_CRON_TOKEN') !== $token) {
+            echo 'Access denied!';
+
+            return false;
+        }
+
+        if (!$this->configuration->getBoolean('SMAILY_ENABLE_ABANDONED_CART')) {
+            echo 'Abandoned cart disabled!';
+
+            return false;
+        }
+
+        if (empty($this->api)) {
+            echo 'Account setup not finished!';
+
+            return false;
+        }
+
+        $count = 0;
+        $failCount = 0;
+        $success = true;
+
+        $abandonedCartsCollection = new AbandonedCartCollection();
+        foreach ($abandonedCartsCollection->carts as $cart) {
+            // Notice!
+            // When running cron job in the same browser the session is shared
+            // and the abandoned cart `date_upd` time is renewed on each request.
+            // This doesn't allow the check to pass. I recommend running the cron job
+            // in a private window to avoid this.
+            if (!$this->isDelayTimePassed($cart)) {
+                break;
+            }
+
+            if (!$this->send($cart)) {
+                ++$failCount;
+                $success = false;
+            }
+
+            ++$count;
+        }
+
+        echo sprintf("%s abandoned cart email(s) sent!\r\n", $count);
+        if (!$success) {
+            echo sprintf("%s failed cart(s), check logs!\r\n", $failCount);
+        }
+
+        return $success;
+    }
+
+    private function isDelayTimePassed(AbandonedCart $cart): bool
+    {
+        $cart_updated_time = strtotime($cart->dateUpdated);
+        $syncInterval = $this->configuration->get('SMAILY_ABANDONED_CART_TIME');
+
+        $reminder_time = strtotime('+' . $syncInterval . ' minutes', $cart_updated_time);
+        $current_time = strtotime(date('Y-m-d H:i') . ':00');
+        // Don't continue if cart delay time has not passed.
+        if ($current_time < $reminder_time) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Send abandoned cart emails to customers.
+     *
+     * @return void
+     */
+    private function send(AbandonedCart $cart): bool
+    {
+        $response = $this->api->triggerAutomation(
+            $this->configuration->get('SMAILY_CART_AUTORESPONDER'),
+            [$this->generatePayload($cart)],
+        );
+
+        if ($response->getStatusCode() !== 200) {
+            Logger::logErrorWithFormatting(
+                'Failed sending out abandoned cart email for email: %s, cart_id: %s. ' .
+                'Smaily response HTTP response code: %s.',
+                $cart->email,
+                $cart->cartID,
+                $response->getStatusCode()
+            );
+
+            return false;
+        }
+
+        $body = json_decode($response->getBody()->getContents(), true);
+        if (!isset($body['code']) || $body['code'] !== 101) {
```

Comment:
Just a sugar, but mb you will like it more:
if (($body['code'] ?? null) !== 101) {
...
}